src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -411,16 +409,35 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, DiagnosticBag diag
                         var mods = singleDeclaration.Modifiers;
                         if ((mods & DeclarationModifiers.Partial) == 0)
                         {
-                            diagnostics.Add(ErrorCode.ERR_MissingPartial, singleDeclaration.NameLocation, self.Name);
+                            diagnostics.Add(ErrorCode.ERR_MissingPartial, singleDeclaration.NameLocation, this.Name);
                             modifierErrors = true;
                         }
                     }
                 }
             }
 
+            if (this.Name == SyntaxFacts.GetText(SyntaxKind.RecordKeyword))
+            {
+                foreach (var decl in declaration.Declarations)
+                {
+                    var location = decl.NameLocation;
+                    var nameText = location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start).ToString();
```

Comment:
>location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start).ToString(); [](start = 35, length = 78)

This seems too heavyweight. Could we use direct access instead? Perhaps `((decl as SingleTypeDeclaration)?.SyntaxReference.GetSyntax() as BaseTypeDeclarationSyntax)?.Identifier.ValueText`.

Reply:
Or change the loop to loop over `SyntaxReferences`:
```
foreach (var syntaxRef in SyntaxReferences)
{
    var location = syntaxRef.GetLocation();
    var nameText = (syntaxRef as BaseTypeDeclarationSyntax)?.Identifier.ValueText;
    ...
}
```

---
In reply to: [478653015](https://github.com/dotnet/roslyn/pull/47094#discussion_r478653015) [](ancestors = 478653015)